extensions/cornerstone/src/utils/measurementServiceMappings/utils/getSOPInstanceAttributes.js

```diff
@@ -7,12 +7,24 @@ import * as cornerstone from '@cornerstonejs/core';
  * @param {string} imageId The image id of the referenced image
  * @returns
  */
-export default function getSOPInstanceAttributes(imageId) {
+export default function getSOPInstanceAttributes(imageId, displaySetService, annotation) {
   if (imageId) {
     return _getUIDFromImageID(imageId);
   }
 
-  // Todo: implement for volume viewports and use the referencedSeriesInstanceUID
+  const { metadata } = annotation;
+  const { volumeId } = metadata;
+
+  const displaySet = displaySetService.getDisplaySetsBy(displaySet =>
+    volumeId.includes(displaySet.uid)
+  )[0];
```

Comment:
what happens for stack viewport that doesn't have volumeId, this logic doesn't look correct to me

Reply:
Stack shouldnt make it down to this code, it would have the reference imageId, only volume viewport is expected to go down

Reply:
resolved the rest, for stack I'm assuming referencedImageId should always be present, only volume viewport is making it down to this.